packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -692,8 +701,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             },
         );
 
+        // back-compat before namespaces were improved
```

Comment:
i'd like to see tests for this. likly easier to test if was encapsulated in a function, maybe a factory for the datastore object

Reply:
Extracted to function and added a few tests